File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/auth/login/LoginModule.kt
package au.com.crazybean.mobilex.kurir.modules.auth.login

import au.com.crazybean.mobilex.kurir.extension.qualifier
import au.com.crazybean.mobilex.kurir.modules.base.ViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val loginModule = module {
    val qualifier = qualifier<LoginWrapper>()

    // Wrapper via ViewModel
    viewModel(qualifier) {
        ViewModel(LoginWrapper(get(), get()))
    }

    // Actor
    factory { (scene: LoginScene?) ->
        val viewModel = get<ViewModel<LoginWrapper>>(qualifier)
        LoginActor(scene, viewModel.wrapper)
    }
}

File: ios/Kurir/modules/auth/login/LoginModule.swift
//
//  LoginModule.swift
//  Kurir
//
//  Created by Loren on 19/8/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import Mobilex

class LoginModule: Module {
    override func inject() {
        // Wrapper
        factory(LoginWrapper.self) { r in
            LoginWrapper(userData: r.resolve(UserData.self), repository: r.resolve(UsersRepository.self)!)
        }
        
        // Actor
        factory(LoginActor.self) { (r, scene: LoginScene) -> LoginActor in
            LoginActor(scene: scene, wrapper: r.resolve(LoginWrapper.self)!)
        }
    }
}
